src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.cs

```diff
@@ -1108,14 +1115,17 @@ protected override void ValidateAddedValue(IJsonTypeInfoResolver item)
 
             protected override void OnCollectionModifying()
             {
-                _options.VerifyMutable();
+                _options?.VerifyMutable();
             }
 
             protected override void OnCollectionModified()
             {
                 // Collection modified by the user: replace the main
                 // resolver with the resolver chain as our source of truth.
-                _options._typeInfoResolver = this;
+                if (_options is not null)
+                {
+                    _options._typeInfoResolver = this;
```

Comment:
I assume any concurrent use of this is misuse?

Reply:
Correct. No guarantees are provided while the JSO is still in edit mode.